src/convert/metadataConverter.ts

```diff
@@ -63,20 +63,22 @@ export class MetadataConverter {
             if (destructiveChangesTypes.length) {
               // for each of the destructive changes in the component set, convert and write the correct metadata
               // to each manifest
-              destructiveChangesTypes.map((destructiveChangesType) => {
-                const file = this.getDestructiveManifest(destructiveChangesType);
-                const destructiveManifestContents = cs.getPackageXml(4, destructiveChangesType);
-                const destructiveManifestPath = join(packagePath, file);
-                tasks.push(promises.writeFile(destructiveManifestPath, destructiveManifestContents));
-              });
+              tasks.push(
+                ...destructiveChangesTypes.map(async (destructiveChangesType) =>
+                  promises.writeFile(
+                    join(packagePath, this.getDestructiveManifest(destructiveChangesType)),
+                    await cs.getPackageXml(4, destructiveChangesType)
+                  )
+                )
+              );
```

Comment:
I'd still rather see this use a `for ... of` loop. It's much more straight forward than using `.map` + array destructuring and accomplishes the same task.

```suggestion
              for (const destructiveChangesType of destructiveChangesTypes) {
                tasks.push(
                  promises.writeFile(
                    join(packagePath, this.getDestructiveManifest(destructiveChangesType)),
                    await cs.getPackageXml(4, destructiveChangesType)
                  );
                );
              }
```